### ISSUE TITLE
Bug in Image File Path Construction Leading to FileNotFoundError in PascalVOC Dataset Class

### ISSUE THREAD

**Issue by DimaBir** [Deci-AI/super-gradients#1769]
https://github.com/Deci-AI/super-gradients/blob/f1f4d61acba7e8d532798697079b102055fce44a/src/super_gradients/training/datasets/detection_datasets/pascal_voc_detection.py#L98C58-L98C58

- **Issue**: In the `PascalVOCDetectionDataset` class, specifically in the `_setup_data_source` method, there's a line intended to find all `.jpg` image files using `glob`. However, the line:

    ```python
    img_files = glob.glob(img_files_folder + "*.jpg")
    ```

    is missing a forward slash (`/`) between the directory and the wildcard pattern. This results in an incorrect path, causing the system to fail to find any image files, leading to `FileNotFoundError`.

- **Expected Behavior**: The system should correctly identify and list all `.jpg` files in the `img_files_folder`.

- **Impact**: This bug is critical as it prevents the successful loading of image datasets, essential for training and evaluating object detection models on the PASCAL VOC datasets.

### Steps to Reproduce
1. Initialize the `PascalVOCDetectionDataset` class with a valid `data_dir` pointing to the PASCAL VOC dataset.
2. Invoke a method or process that leads to the execution of `_setup_data_source`.
3. Observe the `FileNotFoundError` due to the system's inability to find `.jpg` files in the specified directory.

### Suggested Fix
Use `os.path.join` for proper path construction, which ensures the correct formation of file paths:

```python
img_files = glob.glob(os.path.join(img_files_folder, "*.jpg"))
```

This change guarantees the correct concatenation of the directory path and the file pattern, enabling `glob` to properly locate and list the image files.

**Comment by DimaBir** [comment]
My goal is to create a custom DS that uses the VOC2010 Challenge dataset for object detection and to try to train the model using Hydra recipes.
 
Here is an example of my custom recipe, which I'm using for `dataset_params`. As you can see here the `images_sub_directory` is simply the name of the images directory, without the trailing `/`. So, when `data_dir` is concatenated with `images_sub_directory` we get a path without trailing `/`.

```yaml
root: ${project_dir:}/../data
train_dataset_params:
  dataset: VOC2010DetectionDataset
  root_dir: ${dataset_params.root}
  data_dir: ${dataset_params.root}/VOCdevkit/VOC2010
  download: True
  images_sub_directory: JPEGImages
  transforms:
    - Resize:
        size: [480, 640]
    - RandomHorizontalFlip
    - ToTensor
    - Normalize:
        mean: [0.485, 0.456, 0.406]
        std: [0.229, 0.224, 0.225]
```
